### ISSUE TITLE
[SEQAN 2.3.0] Tests failed on PPC64

### ISSUE THREAD

**Issue by sagitter** [seqan/seqan#1979]
Hi all.

`app_test_ngs_roi` `app_test_samcat` `app_test_yara` are failed on PPC64:

```
      Start  8: app_test_ngs_roi
8: Test command: /usr/bin/python2 "/builddir/build/BUILD/seqan-seqan-v2.3.0/apps/ngs_roi/tests/run_tests.py" "/builddir/build/BUILD/seqan-seqan-v2.3.0" "/builddir/build/BUILD/seqan-seqan-v2.3.0/build/Release"
8: Test timeout computed to be: 9.99988e+06
8: --- stdout begin --
8: Return code of command "/builddir/build/BUILD/seqan-seqan-v2.3.0/build/Release/bin/bam2roi --strand-specific -if /builddir/build/BUILD/seqan-seqan-v2.3.0/apps/ngs_roi/tests/micro_rna_sorted_2l.bam -of /tmp/tmp0RBkqd/out_mrna_2l_ss.roi" was -6.
8: 
8: --- stdout end --
8: -- stderr begin --
8: 
8: -- stderr end --
8: --- stdout begin --
8: Return code of command "/builddir/build/BUILD/seqan-seqan-v2.3.0/build/Release/bin/bam2roi -if /builddir/build/BUILD/seqan-seqan-v2.3.0/apps/ngs_roi/tests/micro_rna_sorted_2l.bam -of /tmp/tmp0RBkqd/out_mrna_2l.roi" was -6.
8: 
8: --- stdout end --
8: -- stderr begin --
8: 
8: -- stderr end --
8: DELETING TEMP DIR /tmp/tmp0RBkqd
8: Executing test for roi_intersect
8: ================================
8: 
8: roi_feature_projection -m projection -ss -ir /builddir/build/BUILD/seqan-seqan-v2.3.0/apps/ngs_roi/tests/small.roi -if /builddir/build/BUILD/seqan-seqan-v2.3.0/apps/ngs_roi/tests/small.bed -or /tmp/tmp0RBkqd/out_small_bed_mprojection_ss.roi OK
8: roi_feature_projection -m projection  -ir /builddir/build/BUILD/seqan-seqan-v2.3.0/apps/ngs_roi/tests/small.roi -if /builddir/build/BUILD/seqan-seqan-v2.3.0/apps/ngs_roi/tests/small.bed -or /tmp/tmp0RBkqd/out_small_bed_mprojection.roi OK
8: roi_feature_projection -m projection -ss -ir /builddir/build/BUILD/seqan-seqan-v2.3.0/apps/ngs_roi/tests/small.roi -if /builddir/build/BUILD/seqan-seqan-v2.3.0/apps/ngs_roi/tests/small.gff -or /tmp/tmp0RBkqd/out_small_gff_mprojection_ss.roi OK
8: roi_feature_projection -m projection  -ir /builddir/build/BUILD/seqan-seqan-v2.3.0/apps/ngs_roi/tests/small.roi -if /builddir/build/BUILD/seqan-seqan-v2.3.0/apps/ngs_roi/tests/small.gff -or /tmp/tmp0RBkqd/out_small_gff_mprojection.roi OK
8: roi_feature_projection -m projection -ss -ir /builddir/build/BUILD/seqan-seqan-v2.3.0/apps/ngs_roi/tests/small.roi -if /builddir/build/BUILD/seqan-seqan-v2.3.0/apps/ngs_roi/tests/small.gtf -or /tmp/tmp0RBkqd/out_small_gtf_mprojection_ss.roi OK
8: roi_feature_projection -m projection  -ir /builddir/build/BUILD/seqan-seqan-v2.3.0/apps/ngs_roi/tests/small.roi -if /builddir/build/BUILD/seqan-seqan-v2.3.0/apps/ngs_roi/tests/small.gtf -or /tmp/tmp0RBkqd/out_small_gtf_mprojection.roi OK
8: roi_feature_projection -m union -ss -ir /builddir/build/BUILD/seqan-seqan-v2.3.0/apps/ngs_roi/tests/small.roi -if /builddir/build/BUILD/seqan-seqan-v2.3.0/apps/ngs_roi/tests/small.bed -or /tmp/tmp0RBkqd/out_small_bed_munion_ss.roi OK
8: roi_feature_projection -m union  -ir /builddir/build/BUILD/seqan-seqan-v2.3.0/apps/ngs_roi/tests/small.roi -if /builddir/build/BUILD/seqan-seqan-v2.3.0/apps/ngs_roi/tests/small.bed -or /tmp/tmp0RBkqd/out_small_bed_munion.roi OK
8: roi_feature_projection -m union -ss -ir /builddir/build/BUILD/seqan-seqan-v2.3.0/apps/ngs_roi/tests/small.roi -if /builddir/build/BUILD/seqan-seqan-v2.3.0/apps/ngs_roi/tests/small.gff -or /tmp/tmp0RBkqd/out_small_gff_munion_ss.roi OK
8: roi_feature_projection -m union  -ir /builddir/build/BUILD/seqan-seqan-v2.3.0/apps/ngs_roi/tests/small.roi -if /builddir/build/BUILD/seqan-seqan-v2.3.0/apps/ngs_roi/tests/small.gff -or /tmp/tmp0RBkqd/out_small_gff_munion.roi OK
8: roi_feature_projection -m union -ss -ir /builddir/build/BUILD/seqan-seqan-v2.3.0/apps/ngs_roi/tests/small.roi -if /builddir/build/BUILD/seqan-seqan-v2.3.0/apps/ngs_roi/tests/small.gtf -or /tmp/tmp0RBkqd/out_small_gtf_munion_ss.roi OK
8: roi_feature_projection -m union  -ir /builddir/build/BUILD/seqan-seqan-v2.3.0/apps/ngs_roi/tests/small.roi -if /builddir/build/BUILD/seqan-seqan-v2.3.0/apps/ngs_roi/tests/small.gtf -or /tmp/tmp0RBkqd/out_small_gtf_munion.roi OK
8: roi_feature_projection -m difference -ss -ir /builddir/build/BUILD/seqan-seqan-v2.3.0/apps/ngs_roi/tests/small.roi -if /builddir/build/BUILD/seqan-seqan-v2.3.0/apps/ngs_roi/tests/small.bed -or /tmp/tmp0RBkqd/out_small_bed_mdifference_ss.roi OK
8: roi_feature_projection -m difference  -ir /builddir/build/BUILD/seqan-seqan-v2.3.0/apps/ngs_roi/tests/small.roi -if /builddir/build/BUILD/seqan-seqan-v2.3.0/apps/ngs_roi/tests/small.bed -or /tmp/tmp0RBkqd/out_small_bed_mdifference.roi OK
8: roi_feature_projection -m difference -ss -ir /builddir/build/BUILD/seqan-seqan-v2.3.0/apps/ngs_roi/tests/small.roi -if /builddir/build/BUILD/seqan-seqan-v2.3.0/apps/ngs_roi/tests/small.gff -or /tmp/tmp0RBkqd/out_small_gff_mdifference_ss.roi OK
8: roi_feature_projection -m difference  -ir /builddir/build/BUILD/seqan-seqan-v2.3.0/apps/ngs_roi/tests/small.roi -if /builddir/build/BUILD/seqan-seqan-v2.3.0/apps/ngs_roi/tests/small.gff -or /tmp/tmp0RBkqd/out_small_gff_mdifference.roi OK
8: roi_feature_projection -m difference -ss -ir /builddir/build/BUILD/seqan-seqan-v2.3.0/apps/ngs_roi/tests/small.roi -if /builddir/build/BUILD/seqan-seqan-v2.3.0/apps/ngs_roi/tests/small.gtf -or /tmp/tmp0RBkqd/out_small_gtf_mdifference_ss.roi OK
8: roi_feature_projection -m difference  -ir /builddir/build/BUILD/seqan-seqan-v2.3.0/apps/ngs_roi/tests/small.roi -if /builddir/build/BUILD/seqan-seqan-v2.3.0/apps/ngs_roi/tests/small.gtf -or /tmp/tmp0RBkqd/out_small_gtf_mdifference.roi OK
8: roi_feature_projection -m intersection -ss -ir /builddir/build/BUILD/seqan-seqan-v2.3.0/apps/ngs_roi/tests/small.roi -if /builddir/build/BUILD/seqan-seqan-v2.3.0/apps/ngs_roi/tests/small.bed -or /tmp/tmp0RBkqd/out_small_bed_mintersection_ss.roi OK
8: roi_feature_projection -m intersection  -ir /builddir/build/BUILD/seqan-seqan-v2.3.0/apps/ngs_roi/tests/small.roi -if /builddir/build/BUILD/seqan-seqan-v2.3.0/apps/ngs_roi/tests/small.bed -or /tmp/tmp0RBkqd/out_small_bed_mintersection.roi OK
8: roi_feature_projection -m intersection -ss -ir /builddir/build/BUILD/seqan-seqan-v2.3.0/apps/ngs_roi/tests/small.roi -if /builddir/build/BUILD/seqan-seqan-v2.3.0/apps/ngs_roi/tests/small.gff -or /tmp/tmp0RBkqd/out_small_gff_mintersection_ss.roi OK
8: roi_feature_projection -m intersection  -ir /builddir/build/BUILD/seqan-seqan-v2.3.0/apps/ngs_roi/tests/small.roi -if /builddir/build/BUILD/seqan-seqan-v2.3.0/apps/ngs_roi/tests/small.gff -or /tmp/tmp0RBkqd/out_small_gff_mintersection.roi OK
8: roi_feature_projection -m intersection -ss -ir /builddir/build/BUILD/seqan-seqan-v2.3.0/apps/ngs_roi/tests/small.roi -if /builddir/build/BUILD/seqan-seqan-v2.3.0/apps/ngs_roi/tests/small.gtf -or /tmp/tmp0RBkqd/out_small_gtf_mintersection_ss.roi OK
8: roi_feature_projection -m intersection  -ir /builddir/build/BUILD/seqan-seqan-v2.3.0/apps/ngs_roi/tests/small.roi -if /builddir/build/BUILD/seqan-seqan-v2.3.0/apps/ngs_roi/tests/small.gtf -or /tmp/tmp0RBkqd/out_small_gtf_mintersection.roi OK
8: bam2roi --strand-specific -if /builddir/build/BUILD/seqan-seqan-v2.3.0/apps/ngs_roi/tests/micro_rna_sorted_2l.bam -of /tmp/tmp0RBkqd/out_mrna_2l_ss.roi FAILED
8: bam2roi -if /builddir/build/BUILD/seqan-seqan-v2.3.0/apps/ngs_roi/tests/micro_rna_sorted_2l.bam -of /tmp/tmp0RBkqd/out_mrna_2l.roi FAILED
8: ==============================
8:      total tests: 26
8:     failed tests: 2
8: successful tests: 24
8: ==============================
 8/26 Test  #8: app_test_ngs_roi .................***Failed    1.52 sec
```

`17: Error when trying to compare /builddir/build/BUILD/seqan-seqan-v2.3.0/apps/samcat/tests/ex1_merged.bam to /tmp/tmpXTcIGL/ex1_merged.bam: CRC check failed 0xb135639a != 0x9a6335b1L <type 'exceptions.IOError'>`

```
      Start 26: app_test_yara
26: Test command: /usr/bin/python2 "/builddir/build/BUILD/seqan-seqan-v2.3.0/apps/yara/tests/run_tests.py" "/builddir/build/BUILD/seqan-seqan-v2.3.0" "/builddir/build/BUILD/seqan-seqan-v2.3.0/build/Release"
26: Test timeout computed to be: 9.99988e+06
26: md5(/builddir/build/BUILD/seqan-seqan-v2.3.0/apps/yara/tests/gold/adeno-genome.sa.val) == 3620fe89c8b3d6f633d8c98bff282ff3 != 47e9b2d64aa32ec9193522a048ccd12f == md5(/tmp/tmpP1hRFl/adeno-genome.sa.val)
26: md5(/builddir/build/BUILD/seqan-seqan-v2.3.0/apps/yara/tests/gold/adeno-genome.lf.drv) == 046f1e35be70da83b057e0a3d8a10a66 != d5e2bc49e202e5427889f0a4ad684787 == md5(/tmp/tmpP1hRFl/adeno-genome.lf.drv)
26: md5(/builddir/build/BUILD/seqan-seqan-v2.3.0/apps/yara/tests/gold/adeno-genome.sa.len) == e48919d8193c5d31a8672baa8cd09924 != 0aa49787b9e067d06baa918e9bdc5fab == md5(/tmp/tmpP1hRFl/adeno-genome.sa.len)
26: md5(/builddir/build/BUILD/seqan-seqan-v2.3.0/apps/yara/tests/gold/adeno-genome.rid.limits) == 03e3c067c21080ea2ec6dedeed7de03b != af367076352443edfbe9acd7951755bc == md5(/tmp/tmpP1hRFl/adeno-genome.rid.limits)
26: md5(/builddir/build/BUILD/seqan-seqan-v2.3.0/apps/yara/tests/gold/adeno-genome.lf.drp) == 8a00080cae8aade040af1c941493bef9 != 10c9bcf2ec72bde87f1e5ade5d9093b4 == md5(/tmp/tmpP1hRFl/adeno-genome.lf.drp)
26: md5(/builddir/build/BUILD/seqan-seqan-v2.3.0/apps/yara/tests/gold/adeno-genome.txt.concat) == e4ced945f802f27e059bb19428b926c0 != 878990e71b9969d84794b7b11c52e695 == md5(/tmp/tmpP1hRFl/adeno-genome.txt.concat)
26: md5(/builddir/build/BUILD/seqan-seqan-v2.3.0/apps/yara/tests/gold/adeno-genome.txt.limits) == 00f4723c47a032a19a625a5fd0aaa920 != b9a441c75b9ff8972edb6f2b7eb35869 == md5(/tmp/tmpP1hRFl/adeno-genome.txt.limits)
26: md5(/builddir/build/BUILD/seqan-seqan-v2.3.0/apps/yara/tests/gold/adeno-genome.txt.size) == 77e90ee2d38ac9ae249e1d34bebb864c != 86b97158404aa322a6e21ba3b3f068f4 == md5(/tmp/tmpP1hRFl/adeno-genome.txt.size)
26: md5(/builddir/build/BUILD/seqan-seqan-v2.3.0/apps/yara/tests/gold/adeno-genome.lf.pst) == 3f06481dcdc72801826caceae2982de2 != 3ed8d4d1025d2e740d51a5a9f2a41f30 == md5(/tmp/tmpP1hRFl/adeno-genome.lf.pst)
26: md5(/builddir/build/BUILD/seqan-seqan-v2.3.0/apps/yara/tests/gold/adeno-genome.sa.ind) == 8a4d9eec18a2290330a81b8bb8b09af0 != ddf405860d6f699b5d91dc9a03293037 == md5(/tmp/tmpP1hRFl/adeno-genome.sa.ind)
26: --- stdout begin --
26: Return code of command "/builddir/build/BUILD/seqan-seqan-v2.3.0/build/Release/bin/yara_mapper /builddir/build/BUILD/seqan-seqan-v2.3.0/apps/yara/tests/gold/adeno-genome /builddir/build/BUILD/seqan-seqan-v2.3.0/apps/yara/tests/input/adeno-reads_1.fa -o /tmp/tmpP1hRFl/adeno-reads_1.t1.sam --threads 1" was 1.
26: Traceback (most recent call last):
26:   File "/builddir/build/BUILD/seqan-seqan-v2.3.0/util/py_lib/seqan/app_tests.py", line 248, in runTest
26:     print >>sys.stderr, stdout_file.read()
26: AttributeError: 'int' object has no attribute 'read'
26: ERROR (when executing "/builddir/build/BUILD/seqan-seqan-v2.3.0/build/Release/bin/yara_mapper /builddir/build/BUILD/seqan-seqan-v2.3.0/apps/yara/tests/gold/adeno-genome /builddir/build/BUILD/seqan-seqan-v2.3.0/apps/yara/tests/input/adeno-reads_1.fa -o /tmp/tmpP1hRFl/adeno-reads_1.t1.sam --threads 1"): 'int' object has no attribute 'read'
26: DELETING TEMP DIR /tmp/tmpP1hRFl
26: Executing test for Yara
26: ==============================
26: 
26: /builddir/build/BUILD/seqan-seqan-v2.3.0/build/Release/bin/yara_indexer /builddir/build/BUILD/seqan-seqan-v2.3.0/apps/yara/tests/input/adeno-genome.fa -o /tmp/tmpP1hRFl/adeno-genome FAILED
26: /builddir/build/BUILD/seqan-seqan-v2.3.0/build/Release/bin/yara_mapper /builddir/build/BUILD/seqan-seqan-v2.3.0/apps/yara/tests/gold/adeno-genome /builddir/build/BUILD/seqan-seqan-v2.3.0/apps/yara/tests/input/adeno-reads_1.fa -o /tmp/tmpP1hRFl/adeno-reads_1.t1.sam --threads 1 FAILED
26: ==============================
26:      total tests: 2
26:     failed tests: 2
26: successful tests: 0
26: ==============================
26/26 Test #26: app_test_yara ....................***Failed    0.15 sec
```

Fedora 26 ppc64
Full log: https://kojipkgs.fedoraproject.org//work/tasks/101/16930101/build.log

**Comment by rrahn** [comment]
@sagitter thx for reporting. We'll investigate. Can you reproduce that it is always these 3 app tests that fail, or is it more sporadic?

**Comment by sagitter** [comment]
>Can you reproduce that it is always these 3 app tests that fail,

Yes.

**Comment by h-2** [comment]
Hm, this will be hard to figure out. @sagitter Do you have recommendations for virtualizing this, or can you provide remote access to a ppc64 machin?

**Comment by sagitter** [comment]
We have remote accesses to test machines for package maintainers of Fedora only.

Fedora 25 PPC64
Fedora 25 PPC64le (little endian)

**Comment by rrahn** [comment]
Can we have access to these machines?

**Comment by h-2** [comment]
@sagitter PowerPC issues should have been resolved by #2142 The last mips issues should be fixed by #2204 . If you can, please give the current develop branch a shot and re-open if issues on Fedora persist. Thank you!

**Comment by sagitter** [comment]
@h-2 
Okay; let me check.

**Comment by sagitter** [comment]
`app_test_ngs_roi app_test_samcat app_test_yara` are now correctly executed.
`razers*` tests failed on every architecture: [build log](https://kojipkgs.fedoraproject.org//work/tasks/9494/23999494/build.log)

The `razers*` errors seem related to the bug #1076.

**Comment by h-2** [comment]
Thanks for checking! 

> The razers* errors seem related to the bug #1076.

It is actually #2184 which won't be fixed for 2.4.0. These failures only happen in Debug mode, please just build the tests in Release mode or manually add `-DNDEBUG`.

**Comment by sagitter** [comment]
@h-2 
Okay, with `-DNDEBUG` all `razers` tests are passed.
